examples/json/client/main.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/yarpc/yarpc-go"
+	"github.com/yarpc/yarpc-go/scheme/json"
+	"github.com/yarpc/yarpc-go/transport"
+	"github.com/yarpc/yarpc-go/transport/http"
+
+	"golang.org/x/net/context"
+)
+
+type GetRequest struct {
+	Key string `json:"key"`
+}
+
+type GetResponse struct {
+	Value string `json:"value"`
+}
+
+type SetRequest struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type SetResponse struct {
+}
+
+func get(ctx context.Context, c json.Client, k string) (string, error) {
+	var response GetResponse
+	_, err := c.Call(
+		ctx,
+		&json.Request{Procedure: "get", Body: &GetRequest{Key: k}},
+		&response,
```

Comment:
does the response not require some sort of `json.Response` object? How would the caller receive response headers?